braindecode/training/scoring.py

```diff
@@ -295,3 +297,47 @@ def on_epoch_end(self, net, dataset_train, dataset_valid, **kwargs):
                 cached_net, dataset_train, self.y_trues_
             )
         self._record_score(net.history, current_score)
+
+
+def predict_trials(module, dataset, return_targets=True):
+    """Create trialswise predictions (n_trials x n_classes x n_predictions),
+    and optionally also return trialwise labels (n_trials x n_targets) from
```

Comment:
n_targets and n_predictions are conceptually different things? sorry but I get confused by n_trials, something I I would call n_crops etc.

Reply:
`n_predictions` is in time domain and depends on `window_size` and size of the receptive field of the net.
`n_targets` can be very different. It can be a single value in classification / regression tasks, it can be multiple values in multiple discrete target classification as introduced with https://github.com/braindecode/braindecode/pull/267, and it can also be a sequence as in https://github.com/braindecode/braindecode/pull/261.

Why do you get confused by `n_trials`? That is the point, we do not have crops / compute windows at this point. These are actual trials. If you want to generate crop / compute window predictions you would call `.predict()` instead. In `predict_trials` we invert the creation of compute windows (removing potentially overlapping predictions) to obtain trial predictions.

Reply:
ok much clearer 🙏 . Maybe it's worth adding a glossary as we do with MNE https://mne.tools/stable/glossary.html ?